src/algebra/order/upper_lower.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import algebra.order.group.basic
+import order.upper_lower
+
+/-!
+# Algebraic operations on upper/lower sets
+
+Upper/lower sets are preserved under pointwise algebraic operations in ordered groups.
+-/
+
+open function set
+open_locale pointwise
+
+section ordered_comm_monoid
+variables {α : Type*} [ordered_comm_monoid α] {s : set α} {x : α}
+
+@[to_additive] lemma is_upper_set.smul_subset (hs : is_upper_set s) (hx : 1 ≤ x) : x • s ⊆ s :=
+smul_set_subset_iff.2 $ λ y, hs $ le_mul_of_one_le_left' hx
+
+@[to_additive] lemma is_lower_set.smul_subset (hs : is_lower_set s) (hx : x ≤ 1) : x • s ⊆ s :=
+smul_set_subset_iff.2 $ λ y, hs $ mul_le_of_le_one_left' hx
+
+end ordered_comm_monoid
+
+section ordered_comm_group
+variables {α : Type*} [ordered_comm_group α] {s t : set α} {a : α}
```

Comment:
I don't think you need commutativity for many of these?

Reply:
`ordered_group` doesn't exist!